architecture/core_objects/builds.adoc

```diff
@@ -164,3 +165,79 @@ $ osc create -f secret.json
 Once you have the `*Secret*` created, you can add a `PushSecretName` field into
 `Output` section inside the `BuildConfig` and set it to the name of the
 `*Secret*` that you created, in this case `*dockerhub*`.
+
+[#using-private-repositories-for-builds]
+== Using private repositories for builds
+
+If you want to build your application from a private repository you must
+supply valid credentials. Currently only SSH key based authentication is supported.
+The keys used to access that repository live in `$HOME/.ssh/` directory.
+They are usually named `id_dsa.pub`, `id_ecdsa.pub`, `id_ed25519.pub` or `id_rsa.pub`.
+If you don't have any, you can generate them with the following command:
+
+====
+
+----
+$ ssh-keygen -t rsa -C "your_email@example.com"
+----
+====
+
+Once that's done you're going to get two files: one containing your public key (as
+explained above) and one containing a corresponding private key (one of `id_dsa`, `id_ecdsa`,
+`id_ed25519` or `id_rsa`). With both of these in place you should consult your
+source control management (SCM) system's manual on how to upload the public key.
+The private one will be used to access your private repository.
+
+Kubernetes provides the
+https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/secrets.md[Secret]
+resource, which you can use to store your keys. You must create the `*Secret*` first
+before builds can use your ssh key for accessing your private repository.
+The `*data*` field for the `*Secret*` object must contain your private key
+with the value set to the base64-encoded content of that file. For example:
+
+====
+
+----
+$ base64 -w 0 $HOME/.ssh/id_rsa
+6yJodHRwc1ovL2zuZGV4LmRvY21lci5aby92MS8iOnsiYXV0aCI6ImJXWnZhblJwYXpwdVoybGxkR2d4TUE9PSIsImVtYWlsIj8ibWlAbWlmby5zayJ9fQ==
+----
+====
+
+Copy the value returned from `base64` command and paste it into `sshsecert` field
+in `*_secret.json_*` file:
+
+====
+
+----
+{
+  "apiVersion": "v1beta3",
+  "kind": "Secret",
+  "metadata": {
+    "name": "scmsecret"
+  },
+  "data": {
+    "sshsecret": "6yJodHRwc1ovL2zuZGV4LmRvY21lci5aby92MS8iOnsiYXV0aCI6ImJXWnZhblJwYXpwdVoybGxkR2d4TUE9PSIsImVtYWlsIj8ibWlAbWlmby5zayJ9fQ=="
+  }
+}
+
+----
+====
+
+To create the `*Secret*` from a *_secret.json_* file, for example, you can use
```

Comment:
"You can then create the `*Secret*` from the *_secret.json_* file using the following command:"